python/lsst/obs/base/butler_tests.py

```diff
@@ -139,13 +152,17 @@ def _test_exposure(self, name):
             self.skipTest("Skipping %s as requested" % (inspect.currentframe().f_code.co_name))
         exp = self.butler.get(name, self.dataIds[name])
 
-        exp_md = self.butler.get(name + "_md", self.dataIds[name])
+        md_component = ".metadata" if self._is_gen3() else "_md"
```

Comment:
At the time I thought I had to do it because I assumed other obs packages used the obs_base tests. It seems like obs_lsst is the only one (other than possibly other external users). The butler_tests can work out from their own butler that it is gen2 vs gen3. At this point I'm inclined to leave the gen2 support in until all four of our obs packages drop gen2 and then rip it all out.